scripts/compress-outputs.sh

```diff
@@ -42,26 +42,33 @@ set -o pipefail # Check all commands in a pipeline
 #   to users is that of the gzip file anyways and so the overwrite case doesn't
 #   do any harm there either.)
 
-echo "Compressing files in ${INDEX_ROOT}/file/ with zero-length marker for try_files"
-pushd ${INDEX_ROOT}/file/
+function compress_dir_with_touch {
+  echo "Compressing files in $1 with zero-length marker for try_files"
+  pushd $1
+  find . -type f > ../compress-file-list
+  # It's possible there won't be matches, in which case grep will exit with 1,
+  # which we want to eat.
+  { egrep '\.gz$' ../compress-file-list || test $? = 1; } | parallel --halt now,fail=1 'gzip -f {} && touch -r {}.gz {}'
+  egrep -v '\.gz$' ../compress-file-list | parallel --halt now,fail=1 'gzip -f {} && touch -r {}.gz {}'
```

Comment:
Might be worth adding the `|| test` bit here too, in case the repo has only gzip files or no files at all.